presto-docs/src/main/sphinx/functions/array.rst

```diff
@@ -62,6 +62,16 @@ Array Functions
     Sorts and returns the array ``x``. The elements of ``x`` must be orderable.
     Null elements will be placed at the end of the returned array.
 
+.. function:: array_sort(array<T>, function<T,T,bigint>) -> array<T>
+
+    Sorts and returns the ``array`` based on the given comparator ``function``. The comparator will take
+    two arguments representing two elements of the ``array``. It returns a negative ``bigint``, zero,
```

Comment:
You should probably say something that the comparator needs to represent a linear ordering. Otherwise you would need to document the requirements on the comparator (transitiveness, etc. ) as e.g. in Javadoc of `Comparator`. Note: `array_sort_by` avoids any ambiguity, because it delegates to the natural ordering of existing data type.